src/connection.rs

```diff
@@ -111,14 +132,39 @@ impl IntoConnectionInfo for String {
 }
 
 fn url_to_tcp_connection_info(url: url::Url) -> RedisResult<ConnectionInfo> {
+    let host = match url.host() {
+        Some(host) => host.to_string(),
+        None => fail!((ErrorKind::InvalidClientConfig, "Missing hostname")),
+    };
+    let port = url.port().unwrap_or(DEFAULT_PORT);
+    let addr = if url.scheme() == "redis+tls" {
+        #[cfg(feature = "tls")]
+        {
+            if url.fragment() == Some("insecure") {
```

Comment:
Nit: Could do

```rust
let insecure = url.fragment() == Some("insecure");
ConnectionAddr::TcpTls {
                    host,
                    port,
                    insecure,
                }
```

Reply:
Should any fragment other than `insecure` and `None` be an error?

Reply:
> Should any fragment other than `insecure` and `None` be an error?

I guess it makes sense, I just pushed a new commit to address both of your comments.

Reply:
@Marwes I guess I should also add something in the documentation about the new feature flag?